src/components/TokenBalance.tsx

```diff
@@ -87,20 +97,26 @@ function TokenBalance({
         <Image source={{ uri: tokensWithUsdValue[0].imageUrl }} style={styles.tokenImg} />
         <View style={styles.column}>
           <Text style={style} testID={'TotalTokenBalance'}>
-            {localCurrencySymbol}
-            {totalTokenBalanceLocal?.toFormat(2) ?? '-'}
-          </Text>
-          <Text style={styles.tokenBalance}>
-            {formatValueToDisplay(tokenBalance)} {tokensWithUsdValue[0].symbol}
+            {(overrideHideBalance || !hideBalance) && localCurrencySymbol}
```

Comment:
I find the overrideHideBalance prop and these booleans expressions hard to read. It seems like only HomeTokenBalance needs to honor the hideBalance state. So could we just use the selector there and TokenBalance could just take a prop on whether to hideBalance or not?